src/functTest/groovy/com/bmuschko/gradle/docker/DockerWaitContainerFunctionalTest.groovy

```diff
@@ -0,0 +1,99 @@
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.DockerPullImage
+            import com.bmuschko.gradle.docker.tasks.container.DockerCreateContainer
+            import com.bmuschko.gradle.docker.tasks.container.DockerStartContainer
+            import com.bmuschko.gradle.docker.tasks.container.DockerWaitContainer
+
+            task pullImage(type: DockerPullImage) {
+                repository = 'alpine'
+                tag = 'latest'
+            }
+
+            task createContainer(type: DockerCreateContainer){
+                dependsOn pullImage
+                targetImageId { pullImage.repository + ":" + pullImage.tag }
+                cmd 'sh', '-c', 'exit 0'
+            }
+
+            task startContainer(type: DockerStartContainer){
+                dependsOn createContainer
+                targetContainerId {createContainer.getContainerId()}
+            }
+
+            task runContainers(type: DockerWaitContainer){
+                dependsOn startContainer
+                targetContainerId {startContainer.getContainerId()}
+                doLast{
+                    if(exitCode != 0){println "Container failed with exit code \${exitCode}"}
+                    else{println "Container successful"}
+                }
+            }
+        """
+
+        expect:
+        BuildResult result = build('runContainers')
+        result.output.contains("Container successful")
+    }
+
+    def "Wait for a container to fail and get the exit code"() {
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.DockerPullImage
+            import com.bmuschko.gradle.docker.tasks.container.DockerCreateContainer
+            import com.bmuschko.gradle.docker.tasks.container.DockerStartContainer
+            import com.bmuschko.gradle.docker.tasks.container.DockerWaitContainer
+
+            task pullImage(type: DockerPullImage) {
+                repository = 'alpine'
+                tag = 'latest'
+            }
+
+            task createContainer(type: DockerCreateContainer){
+                dependsOn pullImage
+                targetImageId { pullImage.repository + ":" + pullImage.tag }
+                cmd 'sh', '-c', 'exit 1'
+            }
+
+            task startContainer(type: DockerStartContainer){
+                dependsOn createContainer
+                targetContainerId {createContainer.getContainerId()}
+            }
+
+            task runContainers(type: DockerWaitContainer){
+                dependsOn startContainer
+                targetContainerId {startContainer.getContainerId()}
+                doLast{
+                    if(exitCode != 0){println "Container failed with exit code \${exitCode}" }
+                    else{println "Container successful"}
+                }
+            }
+        """
+
+        expect:
+        BuildResult result = build('runContainers')
+        result.output.contains("Container failed")
```

Comment:
We should verify the full error message here as printed by your `println` statement. The actual value should be a regex.